src/data_access/SearchDataAccessObject.java

```diff
@@ -0,0 +1,78 @@
+package data_access;
+
+import entity.SearchChatMessage;
+import entity.SearchRequest;
+import okhttp3.*;
+import use_case.search.SearchDataAccessInterface;
+
+public class SearchDataAccessObject implements SearchDataAccessInterface {
+
+  @Override
+  public void getData(SearchRequest searchRequest) {
+
+    // Replace these placeholders with your actual Elasticsearch URL and API key
+    String esUrl = "";
+    String apiKey = "";
+
+    OkHttpClient client = new OkHttpClient();
+
+    // Define the Elasticsearch index to query
+    String index = "search-chats";
+
+    // Define your Elasticsearch query (in this case, a match query for the "name" field)
+    String query = "{ \"query\": { \"match\": { \"name\": \"1984\" } } }";
```

Comment:
maybe use a json stringifier (see https://github.com/uoftm/csc207project/blob/main/src/data_access/FirebaseMessageDataAccessObject.java#L24C26-L24C85) to build a json object from the searchRequest and put it here

Reply:
Done